torch/csrc/jit/python/pybind_utils.cpp

```diff
@@ -157,8 +157,16 @@ IValue toIValue(py::handle obj, const TypePtr& type, c10::optional<int32_t> N) {
               attrName));
         }
 
-        const auto& contained = py::getattr(obj, attrName.c_str());
-        userObj->setSlot(slot, toIValue(contained, attrType));
+        try {
+          const auto& contained = py::getattr(obj, attrName.c_str());
+          userObj->setSlot(slot, toIValue(contained, attrType));
+        } catch (std::exception& e) {
```

Comment:
Nice change! should we still append message of `e` to new exception for sake of providing user more context about the error?

Reply:
Done.